storage/backup/backup-cli/src/backup_types/epoch_ending/restore.rs

```diff
@@ -290,13 +291,24 @@ impl EpochHistory {
                 li_with_sigs.ledger_info() == &self.epoch_endings[0],
                 "Genesis epoch LedgerInfo info doesn't match.",
             );
-            Ok(())
+        } else if let Some(wp_trusted) = self
+            .trusted_waypoints
+            .get(&li_with_sigs.ledger_info().version())
+        {
+            let wp_li = Waypoint::new_any(li_with_sigs.ledger_info());
```

Comment:
this should be new_boundary I think

Reply:
Why restrictive? It's just an extra check.

Reply:
What if somehow you do wish to trust a fork at a non- epoch boundary waypoinit? I feel not doing the assert doesn't lose you anything, while doing the assert adds no value.

Reply:
I don't think we can fork at non-epoch boundary, but whatever

Reply:
haha why not. I guess in this case, we will somehow have a snapshot here and use it as a new genesis.